src/modules/editor/components/extensions/helpers/MediumDialog.tsx

```diff
@@ -0,0 +1,215 @@
+import * as React from "react";
+
+import { client } from "../../../../apolloClient";
+import { queryAccountIDs, IMedium } from "../../../queryHelpers";
+
+import { Dialog, DialogActions, DialogTitle, DialogContent } from '@rmwc/dialog';
+import { TextField } from '@rmwc/textfield';
+import { Select } from "@rmwc/select";
+import { Button } from '@rmwc/button';
+import { CircularProgress } from '@rmwc/circular-progress';
+
+import { createUseStyles } from "react-jss";
+
+import { snackbarQueue } from '../../../../snackbarQueue';
+import { ContributorsField } from '../../helpers/ContributorsField';
+import { CREATE_MEDIUM_MUTATION } from '../../../queryHelpers';
+import classes from "*.module.css";
+
+
+interface IVariables {
+    title: string,
+    user_id: number,
+    caption?: string,
+    attachment_b64: string,
+    media_type: string,
+    is_featured: boolean
+}
+
+interface IData {
+    id: string
+}
+
+const useStyles = createUseStyles({
+    DialogForm: {
+        display: "flex",
+        flexDirection: "column"
+    },
+    Input: {
+        marginTop: "10px"
+    },
+    UploadButton: {
+        display: "flex",
+        flexDirection: "row"
+    }
+})
+
+interface IProps {
+    open: boolean,
+    onClose: (medium?: IMedium) => any,
+    initialData?: IVariables,
+    isFeatured: boolean
+}
+
+export const MediumDialog: React.FC<IProps> = ({ open, onClose, initialData, isFeatured }) => {
+    const styles = useStyles();
+
+    const [title, setTitle] = React.useState(initialData?.title ?? "");
```

Comment:
Okay...so this is where I'd use a `useReducer` as your state is pretty complicated.

Reply:
I don't think that would be a benefit though. If TS had Rust/Haskell style sum types I'd go for it, but this is simpler than a reducer imo